src/pages/web/reverse.mdx

```diff
@@ -2,25 +2,68 @@
 description: To lookup the name of an address we use a reverse resolution. This allows users to indicate a primary name.
 ---
 
+import { EmbedLink } from '../../components/EmbedLink'
 import { EnsProfile } from '../../components/EnsProfile'
+import { QandA } from '../../components/QandA'
 import { Badge } from '../../components/ui/Badge'
 import { Card } from '../../components/ui/Card'
 
 # Primary Names
 
-We can all agree 42-character long machine-optimized addresses (eg. 0x225...c3b5) are not aesthetically pleasing.
-Fortunately, it is super easy to retrieve a user's preferred name, and this page will show you how.
+:::info
+Primary names are now supported on both Ethereum Mainnet and popular L2s (Base, OP Mainnet, Arbitrum One, Scroll, and Linea). This enables users to have an end-to-end experience with ENS on their preferred L2!
+:::
+
+A "primary name" is the result of a bi-directional relationship between an EVM address and a human-readable ENS name. The two directions are:
+
+1. Name -> Address (forward resolution)
+2. Address -> Name (reverse resolution)
+
+The outcome of this relationship makes it safe for applications to display ENS names instead of EVM addresses, leading to a better user experience.
 
 <Card className="flex items-center justify-center gap-2">
   <span className="font-medium">0xb8c...67d5</span>
   <span>to</span>
   <EnsProfile name="nick.eth" hideAddress />
 </Card>
 
-In order to convert them to human-readable names, we use [the reverse registrar](/registry/reverse).
-The reverse registrar is a smart contract that allows users to register their preferred name, referred to as their "primary name" for simplicity purposes.
+While forward resolution is configured in [Resolvers](/resolvers/quickstart), reverse records are managed by smart contracts called Reverse Registrars which you can [read more about below](#setting-primary-names).
```

Comment:
on ethereum, reverse records are managed by both a resolver and the reverse registrar